src/Backends/REFPROP/REFPROPMixtureBackend.cpp

```diff
@@ -1218,6 +1271,242 @@ phases REFPROPMixtureBackend::GetRPphase()
+	 CoolPropDbl dvdrhoL = -1/POW2(rhoL_mass);
+	 CoolPropDbl dvdrhoV = -1/POW2(rhoV_mass);
+	 
+	 CoolPropDbl dvL_dp = dvdrhoL*SatL->calc_first_saturation_deriv(iDmass, iP, rhoL_mol, rhoV_mol);
+	 CoolPropDbl dvV_dp = dvdrhoV*SatV->calc_first_saturation_deriv(iDmass, iP, rhoL_mol, rhoV_mol); 
+	 CoolPropDbl dhL_dp = SatL->calc_first_saturation_deriv(iHmass, iP, rhoL_mol, rhoV_mol);
+	 CoolPropDbl dhV_dp = SatV->calc_first_saturation_deriv(iHmass, iP, rhoL_mol, rhoV_mol);
+	 CoolPropDbl dxdp_h = (Q()*dhV_dp + (1 - Q())*dhL_dp)/(hL_mass - hV_mass);
+	 CoolPropDbl dvdp_h = dvL_dp + dxdp_h*(1/rhoV_mol - 1/rhoL_mol) + Q()*(dvV_dp - dvL_dp);
+	 return -POW2(rhomass())*dvdp_h;
+       }
+       else{
+	 throw ValueError("These inputs are not supported to calc_first_two_phase_deriv");
+       }
+}
+
+CoolPropDbl REFPROPMixtureBackend::calc_first_two_phase_deriv_splined(parameters Of, parameters Wrt, parameters Constant, CoolPropDbl x_end) {
+	// Note: If you need all three values (drho_dh__p, drho_dp__h and rho_spline), 
+	// you should calculate drho_dp__h first to avoid duplicate calculations.
+
+	bool drho_dh__p = false;
+	bool drho_dp__h = false;
+	bool rho_spline = false;
+
+	if (Of == iDmolar && Wrt == iHmolar && Constant == iP){
+		drho_dh__p = true;
+		if (_drho_spline_dh__constp) return _drho_spline_dh__constp;
+	}
+	else if (Of == iDmass && Wrt == iHmass && Constant == iP){
+		return first_two_phase_deriv_splined(iDmolar, iHmolar, iP, x_end)*POW2(molar_mass());
+	}
+	else if (Of == iDmolar && Wrt == iP && Constant == iHmolar){
+		drho_dp__h = true;
+		if (_drho_spline_dp__consth) return _drho_spline_dp__consth;
+	}
+	else if (Of == iDmass && Wrt == iP && Constant == iHmass){
+		return first_two_phase_deriv_splined(iDmolar, iP, iHmolar, x_end)*molar_mass();
+	}
+	// Add the special case for the splined density
+	else if (Of == iDmolar && Wrt == iDmolar && Constant == iDmolar){
+		rho_spline = true;
+		if (_rho_spline) return _rho_spline;
+	}
+	else if (Of == iDmass && Wrt == iDmass && Constant == iDmass){
+		return first_two_phase_deriv_splined(iDmolar, iDmolar, iDmolar, x_end)*molar_mass();
+	}
+	else{
+		throw ValueError("These inputs are not supported to calc_first_two_phase_deriv");
+	}
+
+       if (_RPcheckTwophase() == false)
+	 throw ValueError("Not two phase fluid: saturation derivative is not possible");
+    
+       shared_ptr<REFPROPMixtureBackend> SatL(new REFPROPMixtureBackend(this->calc_fluid_names())), SatV(new REFPROPMixtureBackend(this->calc_fluid_names()));
+       SatL->set_mole_fractions(this->get_mole_fractions());
+       SatV->set_mole_fractions(this->get_mole_fractions());
+       SatL->update(QT_INPUTS, 0.0, _T); // <--- This looks never effecient,
+       SatV->update(QT_INPUTS, 1.0, _T); //      but just mimics the saturate states of HEOS..
+
+       shared_ptr<REFPROPMixtureBackend> Liq(new REFPROPMixtureBackend(this->calc_fluid_names())), End(new REFPROPMixtureBackend(this->calc_fluid_names()));
```

Comment:
Definitely this won't work. You need to just call methods with given temperature and density

Reply:
I'm not sure...
Strange to say, it seems work so far, but, of course, I will dig more..
So, could you please to check the attached test which I tried for verification?
At least in such a pure fluid, I couldn't find anything wrong yet.
I didn't do the mixture of "Methane&Ethane", because it is slightly inconsistent with the outputs between the REFPROP backend and the HEOS one; is there any good model of mixture to be appropriate benchmarking?
![output](https://user-images.githubusercontent.com/20429669/114997155-85378700-9eda-11eb-8346-7707fe98ae2d.png)
```
from pylab import *
import CoolProp.CoolProp as CP

Medium = "Methane"
# two abstract states based on the REFPROP backend and HEOS backend for each..
REFP = CP.AbstractState("REFPROP", Medium)
HEOS = CP.AbstractState("HEOS", Medium)

T = linspace(91, 189, 50)
dTdPsat_RP = []  #dTdP along saturation boundary by REFPROP
dTdPsat_HS = []  #dTdP along saturation boundary by HEOS
dDdTsat_RP = []  #dDdT along saturation boundary by REFPROP
dDdTsat_HS = []  #dDdT along saturation boundary by HEOS
dHdTsat_RP = []  #dHdT along saturation boundary by REFPROP
dHdTsat_HS = []  #dHdT along saturation boundary by HEOS
dDdPsat_RP = []  #dDdP along saturation boundary by REFPROP
dDdPsat_HS = []  #dDdP along saturation boundary by HEOS
dHdPsat_RP = []  #dHdP along saturation boundary by REFPROP
dHdPsat_HS = []  #dHdP along saturation boundary by HEOS

for Tv in T:
    HEOS.update(CP.QT_INPUTS, 0.1, Tv)
    REFP.update(CP.QT_INPUTS, 0.1, Tv)
    dTdPsat_RP.append( REFP.first_saturation_deriv(CP.iT, CP.iP))
    dTdPsat_HS.append( HEOS.first_saturation_deriv(CP.iT, CP.iP))
    dDdTsat_RP.append( REFP.first_saturation_deriv(CP.iDmolar, CP.iT))
    dDdTsat_HS.append( HEOS.first_saturation_deriv(CP.iDmolar, CP.iT))
    dHdTsat_RP.append( REFP.first_saturation_deriv(CP.iHmolar, CP.iT))
    dHdTsat_HS.append( HEOS.first_saturation_deriv(CP.iHmolar, CP.iT))
    dDdPsat_RP.append( REFP.first_saturation_deriv(CP.iDmolar, CP.iP))
    dDdPsat_HS.append( HEOS.first_saturation_deriv(CP.iDmolar, CP.iP))
    dHdPsat_RP.append( REFP.first_saturation_deriv(CP.iHmolar, CP.iP))
    dHdPsat_HS.append( HEOS.first_saturation_deriv(CP.iHmolar, CP.iP))

_, (ax1,ax2) = subplots(2,1)
ax1.plot( T, dTdPsat_RP)
ax1.plot( T, dTdPsat_HS)
ax1.set_ylabel('$\\left(\\frac{dT}{dp}\\right)_{\\rm sat.}$')
ax2.plot(T, array(dTdPsat_RP)-array(dTdPsat_HS))
ax2.set_ylabel('error of $\\left(\\frac{dT}{dp}\\right)_{\\rm sat.}$')
ax2.set_xlabel('$T~(K)$')

_, (ax1,ax2) = subplots(2,1)
ax1.plot( T, dDdTsat_RP)
ax1.plot( T, dDdTsat_HS)
ax1.set_ylabel('$\\left(\\frac{d\\rho}{dT}\\right)_{\\rm sat.}$')
ax2.plot(T, array(dDdTsat_RP)-array(dDdTsat_HS))
ax2.set_ylabel('error of $\\left(\\frac{d\\rho}{dT}\\right)_{\\rm sat.}$')
ax2.set_xlabel('$T~(K)$')

_, (ax1,ax2) = subplots(2,1)
ax1.plot( T, dHdTsat_RP)
ax1.plot( T, dHdTsat_HS)
ax1.set_ylabel('$\\left(\\frac{dH}{dT}\\right)_{\\rm sat.}$')
ax2.plot(T, array(dHdTsat_RP)-array(dHdTsat_HS))
ax2.set_ylabel('error of $\\left(\\frac{dH}{dT}\\right)_{\\rm sat.}$')
ax2.set_xlabel('$T~(K)$')

_, (ax1,ax2) = subplots(2,1)
ax1.plot( T, dDdPsat_RP)
ax1.plot( T, dDdPsat_HS)
ax1.set_ylabel('$\\left(\\frac{d\\rho}{dp}\\right)_{\\rm sat.}$')
ax2.plot(T, array(dDdPsat_RP)-array(dDdPsat_HS))
ax2.set_ylabel('error of $\\left(\\frac{d\\rho}{dp}\\right)_{\\rm sat.}$')
ax2.set_xlabel('$T~(K)$')

_, (ax1,ax2) = subplots(2,1)
ax1.plot( T, dHdPsat_RP)
ax1.plot( T, dHdPsat_HS)
ax1.set_ylabel('$\\left(\\frac{dH}{dp}\\right)_{\\rm sat.}$')
ax2.plot(T, array(dHdPsat_RP)-array(dHdPsat_HS))
ax2.set_ylabel('error of $\\left(\\frac{dH}{dp}\\right)_{\\rm sat.}$')
ax2.set_xlabel('$T~(K)$')

show()
```